packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -0,0 +1,1214 @@
+///     return Scaffold(
+///       appBar: AppBar(
+///         title: const Text('Pannable Table'),
+///       ),
+///       body: InteractiveViewer(
+///         disableRotation: true,
+///         disableScale: true,
+///         child: _getTable(20, 3),
+///       ),
+///     );
+///   }
+/// }
+/// ```
+/// {@end-tool}
+@immutable
+class InteractiveViewer extends StatefulWidget {
+  /// Create an InteractiveViewer.
+  ///
+  /// The [child] parameter must not be null. The [minScale] paramter must be
+  /// greater than zero.
+  InteractiveViewer({
+    Key key,
+    @required this.child,
+    this.boundaryMargin = EdgeInsets.zero,
+    this.disableRotation = false,
+    this.disableScale = false,
+    this.disableTranslation = false,
+    // These default scale values were eyeballed as reasonable limits for common
+    // use cases.
+    this.maxScale = 2.5,
+    this.minScale = 0.8,
+    this.onDoubleTap,
+    this.onHorizontalDragCancel,
+    this.onHorizontalDragDown,
+    this.onHorizontalDragEnd,
+    this.onHorizontalDragStart,
+    this.onHorizontalDragUpdate,
+    this.onLongPress,
+    this.onLongPressUp,
+    this.onPanCancel,
+    this.onPanDown,
+    this.onPanEnd,
+    this.onPanStart,
+    this.onPanUpdate,
+    this.onScaleEnd,
+    this.onScaleStart,
+    this.onScaleUpdate,
+    this.onTap,
+    this.onTapCancel,
+    this.onTapDown,
+    this.onTapUp,
+    this.onVerticalDragCancel,
+    this.onVerticalDragDown,
+    this.onVerticalDragEnd,
+    this.onVerticalDragStart,
+    this.onVerticalDragUpdate,
+    ValueNotifier<Matrix4> transformationController,
+  }) : assert(child != null),
+       assert(minScale != null),
+       assert(minScale > 0),
+       assert(disableTranslation != null),
+       assert(disableScale != null),
+       assert(disableRotation != null),
+       // TODO(justinmc): Remove this assertion when rotation is enabled.
+       assert(disableRotation == true, 'Set disableRotation to true. This requirement will be removed later when the feature is complete.'),
+       transformationController = transformationController ?? ValueNotifier<Matrix4>(Matrix4.identity()),
+       super(key: key);
+
+  /// A margin for the visible boundaries of the child.
+  ///
+  /// Any transformation that results in the viewport being able to view outside
+  /// of the boundaries will be stopped at the boundary. The boundaries do not
+  /// rotate with the rest of the scene, so they are always aligned with the
+  /// viewport.
+  ///
+  /// To produce no boundaries at all, pass infinite [EdgeInsets], such as
+  /// `EdgeInsets.all(double.infinity)`.
+  ///
+  /// No edge can be NaN.
+  ///
+  /// Defaults to EdgeInsets.zero, which results in boundaries that are the
+  /// exact same size and position as the constraints.
+  final EdgeInsets boundaryMargin;
+
+  /// The Widget to perform the transformations on.
+  ///
+  /// [child] should usually have an intrinsic size. This is used to calculate
```

Comment:
I've since updated the widget so that it handles both cases just like any other widget would.    For `child: Container(width: 100, height: 100)`, you'll get a 100x100 box.  For `child: Container()`, it can expand to the size of the parent, like it would if there was no InteractiveViewer.  Things like boundary work intuitively in both cases.

I'll update these docs.